### PR TITLE
Update prowbazel to install goimports and golangci

### PR DESCRIPTION
**Pull request by bianpengyuan**
This is not quite maintainable though.. @geeknoid @howardjohn have you guys thought about making istio-proxy also use build-tools image? It has some snowflake needs and could increases image size quite a lot. Maybe build prowbazel image from build-tools image?

### CODE REVIEW

**Comment by howardjohn** [comment]
> Maybe build prowbazel image from build-tools image?

I think this is what we want ideally

**Comment by clarketm** [comment]
@bianpengyuan
This change will conflict #2001. `0.5.13` is already published. This should be version `0.5.14`.

File: docker/prowbazel/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 PROJECT = istio-testing
-VERSION ?= 0.5.12
+VERSION ?= 0.5.13
```
Comment (clarketm) [inline docker/prowbazel/Makefile]: Same for the other references.
```suggestion
VERSION ?= 0.5.14
```

**Review by clarketm — CHANGES_REQUESTED** [review]

File: docker/prowbazel/Makefile (hunk 1 above)
Comment (bianpengyuan) [reply docker/prowbazel/Makefile]: Ok, I did not see 0.5.13 in the README file so I used it and I think I've already pushed the image with that tag :). Anyway I updated everything to 0.5.14 now.

Comment (clarketm) [reply docker/prowbazel/Makefile]: 1) Prow pushes the image on `postsubmit`. I don't think it is good practice to manually push versioned images overwriting ones that already exist. 
2) That is my fault for not updating the README, I am retroactively doing it here #2011. The version was clearly defined in `Makefile` if you were to rebase.
3) I am not sure what tags you pushed, but now `0.5.12`, `0.5.13`, and `0.5.14` all reference the same image. As long as the `0.5.14` you pushed has my changes from #2001 it should be fine ... and nothing uses those old images.
image

Comment (bianpengyuan) [reply docker/prowbazel/Makefile]: Oh sorry I don't realize there is a postsubmit that does image push. I thought it was always done manually. I won't push manually, thanks!

Yeah, it should be fine for this updates, only 0.5.14 should be used now and let's merge this PR to make sure the tag has all changes?

File: docker/prowbazel/README.md [hunk 2]
```diff
@@ -56,3 +56,4 @@ Our dependency on this script is because it appropariately writes test job resul
 * 0.5.10: Install cmake in /usr/local/bin
 * 0.5.11: Update clang and libc++ to 9.0 and preload Bazel 1.0.0
 * 0.5.12: Install clang and libc++ from the official releases instead of packages.
+* 0.5.14: Install goimports and golangci-lint
```
Comment (fejta) [inline docker/prowbazel/README.md:60]: I'd recommend using bazel to install and manage goimports and golangci-lint for a bazel image

**Review by fejta — APPROVED** [review]
/hold
Unhold at your leisure

File: docker/prowbazel/Makefile (hunk 1 above)
Comment (fejta) [reply docker/prowbazel/Makefile]: Can we change the postsubmit to avoid overwriting existing images?

Comment (clarketm) [reply docker/prowbazel/Makefile]: @fejta #2014

Comment (clarketm) [reply docker/prowbazel/Makefile]: No problem @bianpengyuan . I was more worried about the image tag overwriting than anything else. Can you rebase, then we can merge and be up-to-date.

**Review by clarketm — APPROVED** [review]

File: docker/prowbazel/Makefile (hunk 1 above)
Comment (bianpengyuan) [reply docker/prowbazel/Makefile]: Thanks! Rebased.

File: docker/prowbazel/README.md (hunk 2 above)
Comment (bianpengyuan) [reply docker/prowbazel/README.md:60]: Ah, thanks for the pointer! I think it makes more sense to use the bazel rules for istio/proxy. When proxy uses an image built from `build-tools`, we can switch to go-lint target from common files. 

But anyway, let's merge this to make everything here in sync with images in gcr, there is no harm to do so.

**Pull request merged**